Refactor authentication and multitenancy documentation some (#2915)
Azure/azure-service-operator: docs/hugo/content/guide/annotations.md

    
Unknown values default to `manage`.

### `serviceoperator.azure.com/credential-from`

> Instructs the operator to use a particular credential for the resource. This credential supersedes any global or
> namespace scoped credentials the operator has configured.

Reviewer: Currently, this reads as though the secret should be in the annotation.

```suggestion
Instructs the operator to read the credential for the resource from the specified secret. This credential supersedes any global or namespace scoped credentials the operator has configured.
```

Author: Done

Allow values are:

> - The name of any secret in the same namespace as the resource

Reviewer 2: Not necessarily should be in the same namespace, they can use the secret in another namespace as well using `Namespace/resourceName` syntax as well

Reviewer 2: If its in the same namespace, then they can only use the credential name without the namespaceName

Reviewer: I think these docs are correct - don't we explicitly prohibit cross namespace secrets?
Maybe that needs to be called out here too.

Author: We just changed this. We used to technically (undocumented) support cross-namespace. I removed that in #2919
